### PR TITLE
server to send Client Secret  for payment

### PR DESCRIPTION
**Pull request by YasserAmro0**
router findClientSecret gets ClientSecret and sends it to the front to handle page payment

### CODE REVIEW

File: src/services/payment.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import Stripe from 'stripe';
+import config from '../config';
+
+const apiKey = config.API_KEY;
+const stripeInstance = new Stripe(apiKey as string, {
+  apiVersion: '2022-11-15',
```
Comment (ahmedmurtaja) [inline src/services/payment.ts:6]: why do we need to specify the version?
```suggestion

```

**Review by ahmedmurtaja — CHANGES_REQUESTED** [review]
Great Job Yasser

**Review by raghdarizq — APPROVED** [review]

File: src/controllers/payment.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { NextFunction, Request, Response } from 'express';
+import getClientSecret from '../services/payment';
+
+const findClientSecret = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const paymentIntent = await getClientSecret();
+    return res.send({
```
Comment (LinaYahya) [inline src/controllers/payment.ts:7]:
```suggestion
    return res.json({

```

File: src/routes/index.ts [hunk 3]
```diff
@@ -1,10 +1,12 @@
 import { Router } from 'express';
 import appointmentsRouter from './appointment';
 import therapistRouter from './therapist';
+import { findClientSecret } from '../controllers';
 
 const router = Router();
 
 router.use('/appointments', appointmentsRouter);
 router.use('/therapists', therapistRouter);
+router.post('/create-payment-intent', findClientSecret);
```
Comment (LinaYahya) [inline src/routes/index.ts:10]:
```suggestion
router.post('/payment-intent', findClientSecret);
```

Comment (LinaYahya) [reply src/routes/index.ts:10]: can we create this route as a user protected route, So we can get user email to send

File: src/services/payment.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import Stripe from 'stripe';
+import config from '../config';
+
+const apiKey = config.API_KEY;
+const stripeInstance = new Stripe(apiKey as string, {
+  apiVersion: '2022-11-15',
+});
+
+const calculateOrderAmount = () => 1400;
```
Comment (LinaYahya) [inline src/services/payment.ts:9]: we need to get  session price as well from the client side

File: src/services/payment.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  apiVersion: '2022-11-15',
+});
+
+const calculateOrderAmount = () => 1400;
+
+const getClientSecret = async () => {
+  const paymentIntent = await stripeInstance.paymentIntents.create({
+    amount: calculateOrderAmount(),
+    currency: 'usd',
+    payment_method_types: ['card'],
```
Comment (LinaYahya) [inline src/services/payment.ts:15]:
```suggestion
    payment_method_types: ['card'],
        receipt_email, //user email from user data (userId)

```

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by MohannadSabaa — APPROVED** [review]

**Review by LinaYahya — APPROVED** [review]
Well Done @YasserAmro0

File: src/controllers/payment.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import { NextFunction, Response } from 'express';
+import getClientSecret from '../services/payment';
```
Comment (ahmedmurtaja) [inline src/controllers/payment.ts:2]: import from index

**Review by ahmedmurtaja — APPROVED** [review]
good job

**Pull request merged**